File: src/cpp/MyClass.cpp
//
// Created by Tobias Hegemann on 26.09.22.
//

#include "MyClass.h"
MyClass::MyClass() {

}
int MyClass::multiply(int a, int b) {
  return a * b;
}
std::string MyClass::sayHello(std::string name) {
  return "Hello " + name;
}
int MyClass::getDefaultRubberbandEngineVersion() {
  auto *rb = new RubberBand::RubberBandStretcher(44100, 1);
  auto version = rb->getEngineVersion();
  delete rb;
  return version;
}

File: src/worklets/MyClass.ts
class MyClass {
    private readonly instance: any;

    constructor(module: EmscriptenModule) {
        // @ts-ignore
        this.instance = new module.MyClass();
        console.log(this.instance)
    }

    multiply(a: number, b: number): number {
        return this.instance.multiply(a, b);
    }

    sayHello(name: string): string {
        return this.instance.sayHello(name);
    }

    getDefaultRubberbandEngineVersion(): number {
        return this.instance.getDefaultRubberbandEngineVersion();
    }
}
export {MyClass}